src/common/arrayUtils.ts

```diff
@@ -0,0 +1,11 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+export function asArray<T>(thing: T | ReadonlyArray<T>): T[] {
+  if (Array.isArray(thing)) {
+    return thing;
+  } else {
+    return [thing as T];
```

Comment:
And even when adding a custom isArray function that does a type guard with the proper generic parameter, I couldn't get rid of this cast...